airbyte-scheduler/persistence/src/test/java/io/airbyte/scheduler/persistence/job_factory/OAuthConfigSupplierTest.java

```diff
@@ -42,16 +47,26 @@
   public void setup() throws JsonValidationException, ConfigNotFoundException, IOException {
     configRepository = mock(ConfigRepository.class);
     trackingClient = mock(TrackingClient.class);
-    oAuthConfigSupplier = new OAuthConfigSupplier(configRepository, false, trackingClient);
+    oAuthConfigSupplier = new OAuthConfigSupplier(configRepository, trackingClient);
     sourceDefinitionId = UUID.randomUUID();
     when(configRepository.getStandardSourceDefinition(any())).thenReturn(new StandardSourceDefinition()
         .withSourceDefinitionId(sourceDefinitionId)
         .withName("test")
-        .withDockerImageTag("dev"));
+        .withDockerImageTag("dev")
+        .withSpec(new ConnectorSpecification()
+            .withAdvancedAuth(new AdvancedAuth()
+                .withAuthFlowType(AuthFlowType.OAUTH_2_0)
+                .withPredicateKey(List.of(CREDENTIALS, "auth_type"))
+                .withPredicateValue("oauth")
+                .withOauthConfigSpecification(new OAuthConfigSpecification()
+                    .withCompleteOauthServerOutputSpecification(Jsons.jsonNode(Map.of(
+                        API_CLIENT, Map.of(
+                            "type", "string",
+                            OAuthConfigSupplier.PATH_IN_CONNECTOR_CONFIG, List.of(CREDENTIALS, API_CLIENT)))))))));
```

Comment:
The `ConnectorSpecification` creation logic can be put into a helper method. This logic is needed in multiple places, and most of them share 90% of the parameters. Currently it is a bit hard to read, especially to tell the differences between difference test cases.

Reply:
good point thanks!